nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/flow/synchronization/StandardVersionedComponentSynchronizer.java

```diff
@@ -2429,6 +2440,24 @@ private <T extends Connectable> Set<T> stopOrTerminate(final Set<T> components,
         return stoppedComponents;
     }
 
+    private void notifyScheduledStateChange(final Connectable component, final FlowSynchronizationOptions synchronizationOptions) {
+        if (component instanceof ProcessorNode) {
+            synchronizationOptions.getCallback().onScheduledStateChange((ProcessorNode) component);
+        } else if (component instanceof Port) {
+            synchronizationOptions.getCallback().onScheduledStateChange((Port) component);
+        }
+    }
+
+    private void notifyScheduledStateChange(final ComponentNode component, final FlowSynchronizationOptions synchronizationOptions) {
+        if (component instanceof ProcessorNode) {
+            synchronizationOptions.getCallback().onScheduledStateChange((ProcessorNode) component);
+        } else if (component instanceof Port) {
+            synchronizationOptions.getCallback().onScheduledStateChange((Port) component);
+        } else if (component instanceof ControllerServiceNode) {
+            synchronizationOptions.getCallback().onScheduledStateChange((Port) component);
```

Comment:
This is not the correct type cast. If component instanceof ControllerServiceNode, you're casting to Port.